### ISSUE TITLE
Mistakes PHP code for JavaScript code from inside a .php file

### ISSUE THREAD

**Issue by still-dreaming-1** [universal-ctags/ctags#826]
I wrote this PHP class as a way of testing universal-ctags:

```
<?                                                                                                                                                                                            
function experiment_class() {                                                                                                                                                                 
    return new experiment_class;                                                                                                                                                              
}                                                                                                                                                                                             
class experiment_class extends o {                                                                                                                                                            
}
```

This is very typical of code I would actually write. Yes, I know, I'm weird. Anyway, all I did was run ctags -R at the project root. Then when I try to lookup `o` where it says `extends o`, it takes me to a JavaScript file instead of o.php where my o class is defined.

**Comment by masatake** [comment]
I don't understand the situation well yet.
Please, let me see the generated tags file.

**Comment by masatake** [comment]
>  it takes me to a JavaScript file instead of o.php where my o class is defined.

Who takes you to the JavaScript?
ctags never takes you anywhere. ctags just generates tags file.
A tool you are using takes you somewhere using the generated tags file.

**Comment by still-dreaming-1** [comment]
Yes, I am using this with Neovim. I see your point that the generated tags file would be more descriptive of what universal ctags is actually doing. I wasn't sure how to report that, but the tags file make sense. I will see if I can create a simple tags file that shows the problem.

**Comment by still-dreaming-1** [comment]
I was wrong. The correct tags are getting generated in the tags file. The problem is with Vim/Neovim. It does not attempt to distinguish which tag by that name is the one you are looking for.

**Comment by vhda** [comment]
Try using `g]` instead of `ctrl-]`.

**Comment by still-dreaming-1** [comment]
@vhda I appreciate the help, but that just brings up a list of tags to choose from. I want Vim to use the context of the cursor to intelligently guess and select the most appropriate tag for me. You know, like what any decent IDE would do. Choosing from a list like that is worse than thinking through where the definition would be and looking it up manually.

**Comment by vhda** [comment]
Vim is dumb. Vim is not an IDE and it will constantly fail to meet your expectations if you **need** an IDE.

That said, in this case universal-ctags will not be able to help you because it provides the necessary information but Vim is not able to natively interpret that information as you would like it to. What you can try is to search for a plugin that implements that functionality for you. Unfortunately, I'm not a JavaScript developer and can't help you further regarding such plugins.

Nevertheless, you can try to read this: `:help include-search`. Maybe you can use Vim's include searching functionality to jump to the class you're looking for the way you would like it to.

Finally, if you can't find a solution you can always try to write one yourself. Maybe you could remap `ctrl-]` to call a function that would obtain the current context and then jump to a tag in the format "my.context.tag", which would work as you would like it to if you use `--extra=+q` option in uctags.

**Comment by still-dreaming-1** [comment]
@vhda I know Vim is not an out of the box IDE, I don't need to be told this. I have been using Vim for many years now. As I previously stated, I have already figured out that Vim is not using context to understand what tags to lookup. I think my not putting this all together until now has more to do with misunderstanding what tags are and how Vim looks them up. Vim does not need to be an IDE to be a little bit smart about how it does this, but the tags format would have to support it. I assumed tags were saved in such a way that any given piece of code would have a unique tag created for it, and Vim would just use some standard way of looking up that unique tag in the tags file. Instead it is more like the tags file just has indexes that are named the same as the code you are looking up and each index has a file and search query associated with it so the matching text can be found. In other words the index is not unique. I assumed it didn't work that way because I gave ctags the benefit of a doubt of not being stupidly useless. It is just common sense that the indexes produced should be unique. To work otherwise makes using tags less efficient and more annoying than just thinking through where the definition of something would be and then using an intelligently configured editor to quickly find that part of the code manually.
